src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -196,11 +198,21 @@ void GetFileRequest::Run() {
                  &response_params);
     return;
   }
+  if (length_ > file_size - offset_) {
```

Comment:
Hmmm, since I don't see any way to check the size of a file on the mobile side, I don't think this check should be here. I think that Core should just send whatever data it has and set the `length` value in the response according to how many bytes it sends

Reply:
Not sure what you mean by the mobile side. The file exists on core in the storage folder so I would check the file size on the fs. From what I understand the `GetFile` would return the bytes between `offset` and `offset+length`. This just checks that `offset+length` doesn't exceed the file_size